app/components/inputs/autoSuggestInput/useDataListOptions.tsx

```
@@ -18,25 +40,31 @@ function getResourcePath(type: DataListType): string {
 
 const useDataListOptions = (dataListType: DataListType) => {
   const [dataListOptions, setDataListOptions] = useState<DataListOptions[]>([]);
-  const resourcePath = getResourcePath(dataListType);
+  /**
+   * In the case of the Gerichtsfinder, we need the previously-selected PLZ to fetch the street names.
+   * However, the AutoSuggest can also be used in a Vorabcheck, so it's not possible to discern the loaderData ahead of time
+   */
+  const loaderData = useLoaderData();
+  const postleitzahl = loaderData?.prunedUserData?.plz as
+    | z.infer<typeof adresseSchema.plz>
+    | undefined;
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const response = await fetch(resourcePath);
-
-        if (response.ok) {
-          const json = await response.json();
-          setDataListOptions(json);
-        }
-      } catch {
+        const dataFetcher = getDataFetcher(dataListType, postleitzahl);
+        const options = await dataFetcher(postleitzahl);
+        setDataListOptions(options);
+      } catch (err: unknown) {
+        // eslint-disable-next-line no-console
+        console.error(err);
```

any special reason to have this `console.error` here?

---

felt kinda weird to send a likely super technical error message to the user, but didn't wanna swallow the error entirely